q2_mlab/learningtask.py

```diff
@@ -56,12 +63,26 @@ def __init__(
         n_repeats,
         distance_matrix=None,
     ):
-        self.distance_matrix = distance_matrix
+        # Add any custom algorithms from entry points
+        for entry_point in self.iter_entry_points():
+            name = entry_point.name
+            method = entry_point.load()
+            self.algorithms.update({name: method})
+
+        self.learner = self.algorithms[algorithm]
+        print(params)
         self.params = json.loads(params)
+        if isinstance(self.learner, Pipeline):
+            # Assumes that the last step in the pipeline is the model:
+            prefix = list(self.learner.named_steps)[-1] + "__"
+            # And adds the prefix of that last step to our param dict's keys
+            # so we can access that step's parameters.
+            newparams = {prefix + key: val for key, val in self.params.items()}
+            self.params = newparams
```

Comment:
Something you may want to consider is the case when someone wants to include parameters in the preprocessing steps in their parameter grid,
e.g., using `sklearn.decomposition.PCA` with both `n_components=2` and `n_components=3`. In addition to say, changing `n_estimators` in Random Forest.
Ideally, the platform would be open to that kind of extension.

Reply:
I think to include any preprocessing steps as part of the hyperparameter tuning, we could define that parameter grid explicitly in `_parameters.py` for each custom model and remove this logic here. 
Otherwise, any parameters for the preprocessing steps can be set in the [definition of the Pipeline](https://github.com/knightlab-analyses/imsms-ml/blob/custom_models/custom_models.py), and then hyperparameter tuning could be run for just the model on the different PCA setups. @dhakim87 what do you think?

Reply:
It seems like you have not provided a way to specify custom parameter grids in `_parameters.py` ?